pkg/registry/core/rest/storage_core.go

```diff
@@ -259,7 +277,7 @@ func (c LegacyRESTStorageProvider) NewLegacyRESTStorage(restOptionsGetter generi
 	if err != nil {
 		return LegacyRESTStorage{}, genericapiserver.APIGroupInfo{}, err
 	}
-
+	restStorage.destroyFuncs = append(restStorage.destroyFuncs, controllerStorage.Controller.Store.DestroyFunc)
```

Comment:
instead of this, iterate through the reststorage map to build these in one spot.

Reply:
Done expect to `serviceRESTStorage`. That's not part of the map so it's added to the list explicitly.